src/main/java/org/jahia/modules/sitemap/utils/Utils.java

```diff
@@ -58,6 +58,29 @@
     private Utils() {
     }
 
+    public static Set<JCRNodeWrapper> getSitemapRoots(RenderContext renderContext, String locale) throws RepositoryException {
+        Set<JCRNodeWrapper> results = new HashSet<>();
```

Comment:
Returning nodes outside the session that retreived them because it will be closed by the doExecute in a finally could be dangerous, also it seem;s only the path of the nodes is used in the JSP so may be we could change the return to only paths ?

Reply:
agree on that, thank you for the comment!